### PR TITLE
Fix eval to compute rouge correctly for rouge_score

### LINKED ISSUE #6808: bart-large-cnn ROUGE-L scores
### Who can help
BART + Summarization @sshleifer

## Information

Model I am using is BART.

The problem arises when:
verifying accuracy numbers of facebook/bart-large-cnn on CNN+Daily Mail. The paper reports R1, R2, RL of 44.16, 21.28, 40.90 but I can get only 44.05, 21.07, 30.62. I used [this](https://github.com/abisee/cnn-dailymail) to make my dataset. Is this expected? 

The tasks I am working on is:
* CNN-Dm summarization task

## To reproduce

1. Follow instructions to download dataset
2. Run with `python run_summarization.py --reference_path=data/cnn_dm/test.target data/cnn_dm/test.source results/test.log`

### PR DESCRIPTION
**Pull request by swethmandava**
Fixes #6808

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hi, for the code quality test to pass you should run `make style` (to apply the style changes) and check with `make quality` (to make sure there is none left) at the root of your `transformers` directory.

File: examples/seq2seq/run_eval.py [hunk 1]
```diff
@@ -142,8 +142,13 @@ def run_generate(verbose=True):
 
     # Compute scores
     score_fn = calculate_bleu if "translation" in args.task else calculate_rouge
-    output_lns = [x.rstrip() for x in open(args.save_path).readlines()]
-    reference_lns = [x.rstrip() for x in open(args.reference_path).readlines()][: len(output_lns)]
+
+    # rouge_score expects \n separated sentences within a summary
```
Comment (sshleifer) [inline examples/seq2seq/run_eval.py:146]: shouldn't this happen inside calculate_rouge_score? It's called many places.

Comment (swethmandava) [reply examples/seq2seq/run_eval.py:146]: sentence splitting depends on whether they are decoded with clean_up_tokenization_spaces=False/True. I can set it as a variable in calculate_rouge_score in that case

Comment (sshleifer) [reply examples/seq2seq/run_eval.py:146]: Yeah that would be awesome!

Comment (swethmandava) [reply examples/seq2seq/run_eval.py]: Done!

**Comment by sshleifer** [comment]
Thanks for the contribution @swethmandava !

I'm going to run this over the weekend to see if metrics have changed.
Then if all goes well I'll merge this on Monday.
If you don't hear from by Tuesday, please ping :)

**Comment by sshleifer** [comment]
I cleaned up and added tests in https://github.com/huggingface/transformers/pull/7410

Metrics look good, let me know what you think of the new code! I will add you as PR coauthor!

**Pull request closed without merging**